src/adapter/optimizations/item_lookup.rs

```diff
@@ -83,10 +94,10 @@ pub(crate) fn resolve_crate_items<'a, V: AsVertex<Vertex<'a>> + 'a>(
                             crate_vertex
                                 .pub_item_kind_index
                                 .free_functions
-                                .iter()
+                                .values()
```

Comment:
All the `.values()` calls here are on hashtables, which have non-deterministic iteration order. This is something we want to avoid, because it makes debugging hard: outcomes are non-reproducible.

Because `cargo-semver-checks` is an OSS project, its priorities are a bit different than corporate software. We're very "understaffed" compared to an equivalent corporate project  with similarly wide use, so maintenance issues (issue triage, debugging, bugfixing etc.) are very costly. We have to prioritize maintainability above performance, or else the project will stall and die.

One option is to use `BTreeMap` instead of hashmap indexes. Another option is to come up with a different index layout. What do you think?

Reply:
We could use an `IndexMap` instead of a `HashMap`, as it preserves insertion order, same as a `Vec` would. We already have `indexmap` as a dependency thanks to `toml_edit`, so it shouldn't affect compilation time.